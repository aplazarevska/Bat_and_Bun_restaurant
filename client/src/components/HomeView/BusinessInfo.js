import React from 'react';
import { Container, Typography } from '@mui/material';

const BusinessInfo = () => {
  // Business info
  const businessInfo = [
    {
      type: "Location",
      content: ["379B South Dowling St,", "Darlinghurst NSW 2010"]
    },
    {
      type: "Contact",
      content: ["0452 453 686"]
    },
    {
      type: "Hours",
      content: ["Mon-Fri 9:30am-3:45pm", "Sat-Sun 10:00am-3:30pm"]
    }
  ];

  return (
    <section id="business-info" data-testid="business-info">
      <Container>
        {businessInfo.map((info, i) => (
          <div key={i}>
            <Typography variant="h3" sx={{fontSize: "1.2em", fontWeight: "700"}}>{info.type}</Typography>
            <ul>
              {info.content.map((detail, i) => (
                <li key={i}>{detail}</li>
              ))}
            </ul>
          </div>
        ))}
      </Container>
    </section>
  )
}

export default BusinessInfo;
