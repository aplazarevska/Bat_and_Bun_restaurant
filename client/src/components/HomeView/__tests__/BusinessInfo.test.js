import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import BusinessInfo from '../BusinessInfo';
import { afterEach, expect } from '@jest/globals';

afterEach(() => {
  cleanup();
});

test("Should render business info component", () => {
  render(<BusinessInfo />);
  const businessInfoElement = screen.getByTestId('business-info');
  expect(businessInfoElement).toBeInTheDocument();
});

test("Matches snapshot", () => {
  const tree = renderer.create(<BusinessInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
