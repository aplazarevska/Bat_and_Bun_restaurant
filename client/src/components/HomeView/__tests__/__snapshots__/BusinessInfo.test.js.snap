// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<section
  data-testid="business-info"
  id="business-info"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-ruvpci-MuiTypography-root"
      >
        Location
      </h3>
      <ul>
        <li>
          379B South Dowling St,
        </li>
        <li>
          Darlinghurst NSW 2010
        </li>
      </ul>
    </div>
    <div>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-ruvpci-MuiTypography-root"
      >
        Contact
      </h3>
      <ul>
        <li>
          0452 453 686
        </li>
      </ul>
    </div>
    <div>
      <h3
        className="MuiTypography-root MuiTypography-h3 css-ruvpci-MuiTypography-root"
      >
        Hours
      </h3>
      <ul>
        <li>
          Mon-Fri 9:30am-3:45pm
        </li>
        <li>
          Sat-Sun 10:00am-3:30pm
        </li>
      </ul>
    </div>
  </div>
</section>
`;
